src/main/java/io/mosip/iiitb/odte/lib/MessageBrokerWrapper.java

```diff
@@ -0,0 +1,146 @@
+            OnDemandAppConfig mbConfig,
+            OnDemandTemplateExtractionConsumerImpl odteConsumer,
+            RSACryptoTool rsaCryptoTool,
+            Logger logger
+    ) {
+        this.logger = logger;
+        String topic = mbConfig.kafkaTopic();
+
+        this.consumer = initializeKafka(
+                mbConfig
+        );
+        DummyConsumerRebalanceListener rebalancedListener = new DummyConsumerRebalanceListener();
+        this.consumer.subscribe(Collections.singletonList(topic), rebalancedListener);
+        this.odteConsumer = odteConsumer;
+        this.rsaCryptoTool = rsaCryptoTool;
+    }
+    public void start() {
+        try {
+            loopIt();
+        } finally {
+            this.consumer.close();
+        }
+    }
+
+    private KafkaConsumer<String, String> initializeKafka(
+            OnDemandAppConfig config
+    ) {
+        Properties props = new Properties();
+        props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, config.kafkaBrokers());
+        props.put(ConsumerConfig.GROUP_ID_CONFIG, config.kafkaGroupId());
+        props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");
+        props.put(
+                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
+                StringDeserializer.class.getName()
+        );
+        props.put(
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
+                StringDeserializer.class.getName()
+        );
+
+        logger.debug(
+                String.format(
+                        "Listening on %s:%s\nTopics = %s\nGroupId = %s\nBrokers = %s\n",
+                        config.kafkaHostname(),
+                        config.kafkaPort(),
+                        config.kafkaTopic(),
+                        config.kafkaGroupId(),
+                        config.kafkaBrokers()
+                )
+        );
+        return new KafkaConsumer<>(props);
+    }
+
+    private void loopIt() {
+        logger.debug("Started listening to events");
+        while (true) {
+            final ConsumerRecords<String, String> consumerRecords =
+                    this.consumer.poll(Duration.ofMillis(100));
+            if (consumerRecords.count() == 0) {
+                continue;
+            }
+
+            for (ConsumerRecord<String, String> record : consumerRecords) {
+                DecryptedOnDemandTemplateRecord drc = parseOnDemandTemplateRecord(record);
+                if (drc != null) {
+                    EventConsumerResponse evtConsumerResponse = this.odteConsumer.processRecord(drc);
+                    logger.debug("Processed a record and got response = {}", evtConsumerResponse);
+                }
+            }
+            this.consumer.commitAsync();
+        }
+    }
+
+    private DecryptedOnDemandTemplateRecord parseOnDemandTemplateRecord(ConsumerRecord<String, String> record) {
+        String jsonMessage = record.value();
+        ObjectMapper objectMapper = new ObjectMapper();
```

Comment:
creating new Object mapper for every object processing is expensive

Reply:
fixed :heavy_check_mark: